### PR TITLE
Add account loader for metadata

### PR DESCRIPTION
**Pull request by ogenstad**
This PR adds an account loader specifically for metadata fields for created_by or updated_by. It will allow for lookups mainly for display labels for accounts within the metadata parts of the GraphQL queries.

It also adds a number of missing tests for previous PRs (when we weren't able to query and get back the correct information for metadata responses within the queries).

## Summary by CodeRabbit

* **New Features**
  * Account metadata fields (created_by, updated_by) now resolve to full account objects including display label and type info.

* **Improvements**
  * Batched, per-request loading for account metadata for better query performance and caching.
  * Consistent metadata shape across nodes, attributes, and relationships, with explicit handling for system and unknown accounts.

* **Tests**
  * Added comprehensive unit and functional tests covering resolution, batching, caching, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: backend/infrahub/graphql/loaders/account.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        real_account_ids = [k for k in keys if k != SYSTEM_USER_ID]
+
+        # Batch load real accounts
+        accounts_by_id: dict[str, dict[str, Any]] = {}
+        if real_account_ids:
+            async with self.db.start_session(read_only=True) as db:
+                nodes = await NodeManager.get_many(
+                    db=db,
+                    ids=real_account_ids,
+                    fields=self.params.fields,
```
Comment (ajtmccarty) [inline backend/infrahub/graphql/loaders/account.py:111]: there might be an issue with including the `fields` in here
if two different parts of the same GraphQL query use this loader and request different fields for the same user ID, then I think they will both end up with the same response instead of different responses for the different sets of fields. it shouldn't be too hard to add/update a test to confirm it.

I'm not sure that best way to handle the response though. It would be nice to have this return a list of `CoreAccount` objects, but that won't work with the system/unknown users. maybe this could return the full `id`, `display_label`, `__typename`, `name`, `label`, `description` dict here every time and then it would be pared down to just the requested fields somewhere within graphene

File: backend/tests/unit/graphql/loaders/test_account.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    db: InfrahubDatabase,
+    default_branch: Branch,
+    register_core_models_schema,
+) -> None:
+    """Verify AccountLoaderParams hash is based on branch and timestamp, not fields.
+
+    The hash should be consistent across different field selections so that
+    loaders with the same branch/timestamp can be reused.
+    """
+    from infrahub.core.timestamp import Timestamp
```
Comment (ajtmccarty) [inline backend/tests/unit/graphql/loaders/test_account.py]: can this import go to the top where it belongs?

**Review by ajtmccarty — APPROVED** [review]

File: backend/tests/unit/graphql/loaders/test_account.py (hunk 2 above)
Comment (ogenstad) [reply backend/tests/unit/graphql/loaders/test_account.py]: Yes of course.

File: backend/infrahub/graphql/loaders/account.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/graphql/loaders/account.py:111]: Good point, I've added some tests to highlight that alternate lookups work.
